src/build.d

```diff
@@ -1019,13 +1013,64 @@ version(Windows)
         if (!vsInstallPath.exists)
             throw new Exception("Could not locate the Visual Studio installation directory");
 
-        const vcVersionFile = vsInstallPath.buildPath("VC", "Auxiliary", "Build", "Microsoft.VCToolsVersion.default.txt");
+        const vcInstallPath = vsInstallPath.buildPath("VC");
+        if (!vcInstallPath.exists)
+            throw new Exception("Could not locate the Visual C++ installation directory");
+
+        return vcInstallPath;
+    }
+
+    /**
+    Sets certain MSVC environment variables
+
+    Params:
+        vcInstallPath = the absolute path to the MSVC installation directory
+        model         = a string describing the host's model, "64" or "32"
+    */
+    void setVCVars(string vcInstallPath, string model)
+    {
+        const vcVarsFilename = "vcvars" ~ model ~ ".bat";
+        const vcVars = vcInstallPath.buildPath("Auxiliary", "Build", vcVarsFilename);
+        if (!vcVars.exists)
+            throw new Exception("Could not locate the Visual C++ " ~ vcVarsFilename ~ " file");
+
+        // run vcvars{arch}.bat and return the resulting environment variables with the `set` command
+        const vsVars = ["cmd", "/C", vcVars, "&&", "set"].execute.output;
+
+        // copy the environment variables we need to the current environment
+        const varsToCopy = ["INCLUDE", "LIBPATH"];
+        foreach(line; vsVars.split("\r\n"))
+        {
+            foreach(var; varsToCopy)
+            {
+                if (line.startsWith(var ~ "="))
+                {
+                    env.getDefault(var, line.split("=")[1]);
+                    break;
+                }
+            }
+        }
+    }
+
+    /**
+    Gets the absolute path to the host's MSVC bin directory
+
+    Params:
+        vcInstallPath = the absolute path to the MSVC installation directory
+        model         = a string describing the host's model, "64" or "32"
+    Returns: a string that is the absolute path to the host's MSVC bin directory
+    */
+    auto getHostMSVCBinDir(string vcInstallPath, string model)
+    {
+        // See https://github.com/Microsoft/vswhere/wiki/Find-VC
+
+        const vcVersionFile = vcInstallPath.buildPath("Auxiliary", "Build", "Microsoft.VCToolsVersion.default.txt");
         if (!vcVersionFile.exists)
             throw new Exception(`Could not locate the Visual C++ version file "%s"`.format(vcVersionFile));
 
         const vcVersion = vcVersionFile.readText().replace("\r\n", "");
         const vcArch = model == "64" ? "x64" : "x86";
-        const vcPath = vsInstallPath.buildPath("VC", "Tools", "MSVC", vcVersion, "bin", "Host" ~ vcArch, vcArch);
+        const vcPath = vcInstallPath.buildPath("Tools", "MSVC", vcVersion, "bin", "Host" ~ vcArch, vcArch);
```

Comment:
BTW: This path is very specific to VS2017, older versions work differently. The auto-tester runs VS2013.

Reply:
What do you suggest?

Reply:
Using vswhere is specific to VS2017 to begin with, so it's ok to build the path to VC this way. It's a lot simpler to read it (up to vcVersion) from the environment variable `VCToolsInstallDir`, though.

If you want to support more than VS2017, I'd rather use the detection in dmd.link.